service/worker/failovermanager/rebalance_workflow_test.go

```diff
@@ -0,0 +1,294 @@
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/resource"
+	"github.com/uber/cadence/common/types"
+)
+
+type rebalanceWorkflowTestSuite struct {
+	suite.Suite
+	testsuite.WorkflowTestSuite
+	activityEnv *testsuite.TestActivityEnvironment
+	workflowEnv *testsuite.TestWorkflowEnvironment
+}
+
+func TestRebalanceWorkflowTestSuite(t *testing.T) {
+	suite.Run(t, new(rebalanceWorkflowTestSuite))
+}
+
+func (s *rebalanceWorkflowTestSuite) SetupTest() {
+	s.activityEnv = s.NewTestActivityEnvironment()
+	s.workflowEnv = s.NewTestWorkflowEnvironment()
+	s.workflowEnv.RegisterWorkflowWithOptions(RebalanceWorkflow, workflow.RegisterOptions{Name: RebalanceWorkflowTypeName})
+	s.workflowEnv.RegisterActivityWithOptions(FailoverActivity, activity.RegisterOptions{Name: failoverActivityName})
+	s.workflowEnv.RegisterActivityWithOptions(GetDomainsForRebalanceActivity, activity.RegisterOptions{Name: getRebalanceDomainsActivityName})
+	s.activityEnv.RegisterActivityWithOptions(FailoverActivity, activity.RegisterOptions{Name: failoverActivityName})
+	s.activityEnv.RegisterActivityWithOptions(GetDomainsForRebalanceActivity, activity.RegisterOptions{Name: getRebalanceDomainsActivityName})
+}
+
+func (s *rebalanceWorkflowTestSuite) TearDownTest() {
+	s.workflowEnv.AssertExpectations(s.T())
+}
+
+func (s *rebalanceWorkflowTestSuite) TestGetDomainsForRebalanceActivity_ReturnOne() {
+	actEnv, mockResource, controller := s.prepareTestActivityEnv()
+	defer controller.Finish()
+	defer mockResource.Finish(s.T())
+
+	domains := &types.ListDomainsResponse{
+		Domains: []*types.DescribeDomainResponse{
+			{
+				DomainInfo: &types.DomainInfo{
+					Name: "d1",
+					Data: map[string]string{
+						common.DomainDataKeyForManagedFailover:  "true",
+						common.DomainDataKeyForPreferredCluster: "c2",
+					},
+				},
+				ReplicationConfiguration: &types.DomainReplicationConfiguration{
+					ActiveClusterName: "c1",
+					Clusters:          clusters,
+				},
+				IsGlobalDomain: true,
+			},
+			{
+				DomainInfo: &types.DomainInfo{
+					Name: "d2",
+					Data: map[string]string{
+						common.DomainDataKeyForManagedFailover:  "false",
+						common.DomainDataKeyForPreferredCluster: "c2",
+					},
+				},
+				ReplicationConfiguration: &types.DomainReplicationConfiguration{
+					ActiveClusterName: "c1",
+					Clusters:          clusters,
+				},
+				IsGlobalDomain: true,
+			},
+			{
+				DomainInfo: &types.DomainInfo{
+					Name: "d3",
+					Data: map[string]string{
+						common.DomainDataKeyForManagedFailover:  "true",
+						common.DomainDataKeyForPreferredCluster: "c1",
+					},
+				},
+				ReplicationConfiguration: &types.DomainReplicationConfiguration{
+					ActiveClusterName: "c1",
+					Clusters:          clusters,
+				},
+				IsGlobalDomain: true,
+			},
+			{
+				DomainInfo: &types.DomainInfo{
+					Name: "d4",
+					Data: map[string]string{
+						common.DomainDataKeyForManagedFailover: "false",
```

Comment:
nit: add a test case where ManagedFailover is true but there's no preferred cluster?